tests/gpu_benchmarking_test.py

```diff
@@ -0,0 +1,198 @@
+        (2) Create a grayscale image of the scattered points with dimensions 32 by 32
+        (3) Perturb the image with an uniform noise with values betweeen 0 and 0.2
+    Parameters
+    ----------
+    parameter : array-like, shape (2)
+        The two input parameters of the model, ordered as [r, theta]
+    return_points : bool (default: False)
+        Whether the simulator should return the coordinates of the simulated data
+        points as well
+
+    Returns
+    -------
+    image: torch tensor, shape (1, 1024)
+        Output flattened image
+    (optional) points: array-like, shape (100, 2)
+        Coordinates of the 2D simulated data points
+    """
+    r, theta = params
+
+    sigma_points = 0.10
+    npoints = 100
+    nx = 32
+    ny = 32
+    sigma_image = 0.20
+
+    points = []
+
+    # Generate points according to params and add noise.
+    points = torch.tensor(
+        [[r * torch.cos(theta), r * torch.sin(theta)]]
+    ) + sigma_points * torch.randn(npoints, 2)
+
+    # Find indices of points within unit circle in image.
+    image = zeros((nx, ny))
+    for point in points:
+        pi = int((point[0] - (-1)) / ((+1) - (-1)) * nx)
+        pj = int((point[1] - (-1)) / ((+1) - (-1)) * ny)
+        if (pi < nx) and (pj < ny):
+            image[pi, pj] = 1
+    # Add uniform noise.
+    image += sigma_image * torch.rand(nx, ny)
+    image = image.T
+    image = image.reshape(1, -1)
+
+    if return_points:
+        return image, points
+    else:
+        return image
+
+
+class CNNEmbedding(nn.Module):
+    """Big CNN embedding net to levarage GPU computation."""
+
+    def __init__(self):
+        super().__init__()
+        # 2D convolutional layer
+        self.conv1 = nn.Conv2d(in_channels=1, out_channels=16, kernel_size=5, padding=2)
+        self.conv2 = nn.Conv2d(
+            in_channels=16, out_channels=64, kernel_size=5, padding=2
+        )
+        self.conv3 = nn.Conv2d(in_channels=64, out_channels=6, kernel_size=5, padding=2)
+        # Maxpool layer that reduces 32x32 image to 4x4
+        self.pool = nn.MaxPool2d(kernel_size=8, stride=8)
+        # Fully connected layer taking as input the 6 flattened output arrays from the
+        # maxpooling layer
+        self.fc = nn.Linear(in_features=6 * 4 * 4, out_features=8)
+
+    def forward(self, x):
+        x = x.view(-1, 1, 32, 32)
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = self.pool(F.relu(self.conv3(x)))
+        x = x.view(-1, 6 * 4 * 4)
+        x = F.relu(self.fc(x))
+        return x
+
+
+class LinearEmbedding(nn.Module):
+    """Naive tall linear network to process 32 by 32 images."""
+
+    def __init__(self):
+        super().__init__()
+        self.l1 = nn.Linear(in_features=32 * 32, out_features=100)
+        self.l2 = nn.Linear(in_features=100, out_features=8)
+
+    def forward(self, x):
+        x = F.relu(self.l1(x))
+        x = F.relu(self.l2(x))
+        return x
+
+
+@pytest.mark.slow
```

Comment:
around 15min (12min CPU, 3min GPU)

But it wouldnt run remotely anyways because there is no GPU, it would fail then.

Reply:
I'm rather concerned about our test-suite taking too long (including slow tests). 15 minutes is quite a lot imo.

Since this is only testing speed, couldn't we reduce num_simulations to 1000 or so? And also `num_epochs` even to 10?

Reply:
Yes, that's definitively possible. we could even reduce this test to just testing that things run through, not testing the speed. The test was meant mainly for benchmarking and I can just move that somewhere else.